cli/cmd/cdk.go

```diff
@@ -97,8 +97,28 @@ func (c *cliState) Honeyvent(ctx context.Context, in *cdk.HoneyventRequest) (*cd
 	return &cdk.Reply{}, nil
 }
 
-// Serve will start the CDK gRPC server at the provided port and log level
-func (c *cliState) Serve(target string) error {
+// Serve will start the CDK gRPC server
+func (c *cliState) Serve() error {
+	// Start the gRPC server for components to communicate back
+	const max_attempts = 5
+
+	if target := os.Getenv("LW_CDK_TARGET"); target != "" {
+		return c.serve(target)
+	}
+
+	// Try a range of port numbers in case the default one is not available
+	var err error
+	for i := 0; i < max_attempts; i++ {
+		err = c.serve(fmt.Sprintf("localhost:%v", defaultGrpcPort+i))
+		if err == nil {
+			c.cdkServerPort = defaultGrpcPort + i
+			return err
+		}
+	}
+	return err
```

Comment:
What about something more meaningful like:
```suggestion
	return errors.Wrap(err, fmt.Sprintf("unable to start gRPC server (attempts: %d)", maxAttempts))
```